src/services/content.ts

```diff
@@ -0,0 +1,344 @@
+ * @param message announcment information from batch
+ * @param activityContent activity content retrieved from announcment
+ * @param blockNumber number of block containing the publication
+ * @param blockIndex index of log message (relative to other DSNP logs) within the block
+ * @param batchIndex index of announcment within the batch
+ */
+const dispatchActivityContent = (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement,
+  activityContent: ActivityContentNote | ActivityContentProfile,
+  blockNumber: number,
+  blockIndex: number,
+  batchIndex: number
+) => {
+  if (isActivityContentNoteType(activityContent)) {
+    return dispatchFeedItem(
+      dispatch,
+      message,
+      activityContent as ActivityContentNote,
+      blockNumber
+    );
+  } else if (isActivityContentProfileType(activityContent)) {
+    return dispatchProfile(
+      dispatch,
+      message,
+      activityContent as ActivityContentProfile,
+      blockNumber,
+      blockIndex,
+      batchIndex
+    );
+  }
+};
+
+/**
+ * dispatchFeedItem dispatches a feed item to the redux store
+ * @param dispatch function used to dispatch to store
+ * @param message announcment information from batch
+ * @param content activity content retrieved from announcment
+ * @param blockNumber number of block containing the publication
+ */
+const dispatchFeedItem = (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement | ReplyAnnouncement,
+  content: ActivityContentNote,
+  blockNumber: number
+) => {
+  const decoder = new TextDecoder();
+  dispatch(
+    addFeedItem({
+      fromId: decoder.decode((message.fromId as any) as Uint8Array),
+      blockNumber: blockNumber,
+      hash: decoder.decode((message.contentHash as any) as Uint8Array),
+      published: content.published,
+      uri: decoder.decode((message.url as any) as Uint8Array),
+      content: content,
+      inReplyTo:
+        message.announcementType === core.announcements.AnnouncementType.Reply
+          ? decoder.decode((message.inReplyTo as any) as Uint8Array)
+          : undefined,
+    })
+  );
+};
+
+/**
+ * dispatchProfile dispatches a profile to the redux store
+ * @param dispatch function used to dispatch to store
+ * @param message announcment information from batch
+ * @param profile profile to dispatch
+ * @param blockNumber number of block containing the publication
+ * @param blockIndex index of log message (relative to other DSNP logs) within the block
+ * @param batchIndex index of announcment within the batch
+ */
+const dispatchProfile = (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement,
+  profile: ActivityContentProfile,
+  blockNumber: number,
+  blockIndex: number,
+  batchIndex: number
+) => {
+  const decoder = new TextDecoder();
+
+  dispatch(
+    upsertProfile({
+      ...profile,
+      fromId: decoder.decode((message.fromId as any) as Uint8Array),
+      blockNumber,
+      blockIndex,
+      batchIndex,
+    } as Profile)
```

Comment:
Why do we need the `as`?

Reply:
This goes away with the SDK 3.0 changes (a PR stacked on this one). It's required, because the parsed values coming out of Parquet don't match the Profile type, so we have to coerce it.